ui/src/pages/execution/Execution.jsx

```diff
@@ -226,9 +226,14 @@ export default function Execution() {
     if (task) {
```

Comment:
Suggestion: Eliminate all logic in `handleSelectedTask`. `task.domain` should be automatically populated by API already.